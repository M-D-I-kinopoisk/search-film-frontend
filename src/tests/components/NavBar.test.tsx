import NavBar from '../../components/Header/NavBar/NavBar'
import { render, screen } from '@testing-library/react'
import { NextIntlProvider } from 'next-intl'

describe('NavBar', () => {
    it('NavBar renders without crashing ', () => {
        const component = render(
            <NextIntlProvider messages={{
                'header': {
                    'title1': 'Мой Иви',
                    'title2': 'Что нового',
                    'title3': 'Фильмы',
                    'title4': 'Сериалы',
                    'title5': 'Мультфильмы',
                    'title6': 'TV+',
                    'title7': 'Оплатить подписку',
                    'title8': 'Поиск'
                },
            }} locale={'ru'}>
                <NavBar
                    handleMouseEnter={jest.fn()}
                    handleMouseLeave={jest.fn()} />
            </NextIntlProvider>
        )
        expect(component).toMatchSnapshot()
        const links = screen.getAllByRole('link')
        expect(links).toHaveLength(6)
    })
})