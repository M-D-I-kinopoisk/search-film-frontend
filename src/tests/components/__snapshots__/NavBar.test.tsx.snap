// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar NavBar renders without crashing  1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar"
      >
        <ul
          class="navbar__ul"
        >
          <li
            class="navbar__li"
          >
            <a
              class="navbar__a"
              href="/"
              title="Мой Иви"
            >
              Мой Иви
            </a>
          </li>
          <li
            class="navbar__li"
          >
            <a
              class="navbar__a"
              href="https://www.ivi.ru/new"
              title="Что нового"
            >
              Что нового
            </a>
          </li>
          <li
            class="navbar__li"
            id="header-movies"
          >
            <a
              class="navbar__a"
              href="/movies"
              id="header-movies"
              title="Фильмы онлайн"
            >
              Фильмы
            </a>
          </li>
          <li
            class="navbar__li"
            id="header-series"
          >
            <a
              class="navbar__a"
              href="https://www.ivi.ru/series"
              id="header-series"
              title="Сериалы онлайн"
            >
              Сериалы
            </a>
          </li>
          <li
            class="navbar__li"
            id="header-animation"
          >
            <a
              class="navbar__a"
              href="https://www.ivi.ru/animation"
              id="header-animation"
              title="Мультфильмы онлайн"
            >
              Мультфильмы
            </a>
          </li>
          <li
            class="navbar__li"
          >
            <a
              class="navbar__a"
              href="https://www.ivi.ru/tvplus"
              id="header-tv"
              title="TV+"
            >
              TV+
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar"
    >
      <ul
        class="navbar__ul"
      >
        <li
          class="navbar__li"
        >
          <a
            class="navbar__a"
            href="/"
            title="Мой Иви"
          >
            Мой Иви
          </a>
        </li>
        <li
          class="navbar__li"
        >
          <a
            class="navbar__a"
            href="https://www.ivi.ru/new"
            title="Что нового"
          >
            Что нового
          </a>
        </li>
        <li
          class="navbar__li"
          id="header-movies"
        >
          <a
            class="navbar__a"
            href="/movies"
            id="header-movies"
            title="Фильмы онлайн"
          >
            Фильмы
          </a>
        </li>
        <li
          class="navbar__li"
          id="header-series"
        >
          <a
            class="navbar__a"
            href="https://www.ivi.ru/series"
            id="header-series"
            title="Сериалы онлайн"
          >
            Сериалы
          </a>
        </li>
        <li
          class="navbar__li"
          id="header-animation"
        >
          <a
            class="navbar__a"
            href="https://www.ivi.ru/animation"
            id="header-animation"
            title="Мультфильмы онлайн"
          >
            Мультфильмы
          </a>
        </li>
        <li
          class="navbar__li"
        >
          <a
            class="navbar__a"
            href="https://www.ivi.ru/tvplus"
            id="header-tv"
            title="TV+"
          >
            TV+
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
